sql/lesson5/tutorial.md

```diff
@@ -0,0 +1,181 @@
+---
+layout: page
+title: Introduction to SQL
+---
+
+## About Lesson 5
+
+The previous lessons showed you how to manipulate data from a single table. _Lesson 5_ will show you how to combine data from multiple tables and will bring out the _relational_ part of relational databases.
+
+Tables in a relational database relate to each other as they contain keys which allow information from multiple tables to be combined easily. The real power of relational databases, and SQL, comes from working with multiple tables at once.
+
+In this lesson we will keep using the same database as before (https://sqliteonline.com/#fiddle-5be888fa749e451ljodase3y)
+
+### Primary and Foreign Keys
+
+All tables in a relational database need to have a primary key. This is an identifier that uniquely identifies each row. We saw in _Lesson 4_ that the `people` table uses the `id` field as a primary key. The key aspects of the primary key field are that it is never empty and it is unique: every record in a table has one and only one record can have a given primary key. For instance, in the `people` table, record with `id` `1` is the record for `Emma Smith`. If we try to insert a new record with the same `id` primary key, it will fail. If we try to insert a record with no value for `id` it will fail too.
+
+Any table can also have a number of foreign key fields. Each foreign key contains the value for a primary key in another table. For example, if we select the data in the `hosts` table:
```

Comment:
Suggest improvement along the lines of:
`A foreign key is a field that contains the primary key value of another table, to indicate which record from that table relates to this row. This is known as a _reference_.`
Suggest concept of multiple foreign keys per table comes later.

Reply:
Done